### PR TITLE
Fix issue 24464

### PR DESCRIPTION
**Pull request by ejwessel**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/24570?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/24464

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by ejwessel** [comment]
It seems https://github.com/MetaMask/metamask-extension/commit/60e4dda4e4cfbda03144b45e39ad19c1d7742687
reintroduced the issue.

File: ui/ducks/send/swap-and-send-utils.ts [hunk 1]
```diff
@@ -132,9 +132,9 @@ const QUOTE_VALIDATORS = [
   },
   {
     property: 'error',
-    type: 'string',
-    validator: (error: string | null) =>
-      error === null || typeof error === 'string',
+    type: 'object',
+    validator: (error: any) =>
```
Comment (BZahory) [inline ui/ducks/send/swap-and-send-utils.ts:136]: Let's use `unknown` here instead

File: ui/ducks/send/swap-and-send-utils.ts [hunk 2]
```diff
@@ -132,9 +132,9 @@ const QUOTE_VALIDATORS = [
...
+      error === null || typeof error === 'object',
```
Comment (BZahory) [inline ui/ducks/send/swap-and-send-utils.ts:137]: The linter is also complaining about a new line here; there may be a broken setting in your VSCode

**Review by BZahory — APPROVED** [review]

**Comment by sleepytanya** [comment]
@ejwessel @BZahory 
Not sure what's wrong, probably it is due to dev build being extremely slow but confirmation is stuck on estimation spinner:

https://github.com/MetaMask/metamask-extension/assets/104780023/ce7fda27-6dd5-445f-bd65-a337304472d3

If you close MM extendede view (or close popup) and open it again the spinner is gone but no 'estimated changes' message is shown.

Screenshot 2024-05-16 at 18 40 22

Token list displays 0 balances (except ETH).


Screenshot 2024-05-16 at 18 38 55

**Comment by sleepytanya** [comment]
It appears if I wait long enough the simulation doesn't fail:
https://jam.dev/c/d9cb3f46-a2b2-4641-943b-a2bed0126efa

**Comment by sleepytanya** [comment]
Just for the tracking - it is a bug, simulations fail if send+swap performed for non-native tokens:

https://jam.dev/c/602ed3dd-eb48-402d-abb3-c69a56f6b122

**Pull request merged**